plasmapy/simulation/tests/test_particletracker.py

```diff
@@ -41,64 +31,79 @@ def sine(t, amplitude, omega, phase, mean):
     return params, stds
 
 
-# def test_particle_uniform_magnetic():
-#     r"""
-#         Tests the particle stepper for a uniform magnetic field motion.
-#     """
-#     test_plasma = uniform_magnetic_field()
+E_unit = u.V / u.m
 
-#     particle_type = 'N-14++'
-#     s = ParticleTracker(test_plasma, particle_type=particle_type, dt=1e-2 * u.s,
-#                 nt=int(1e2))
+# @profile
 
-#     perp_speed = 0.01 * u.m / u.s
-#     parallel_speed = 1e-5 * u.m / u.s
-#     mean_B = test_plasma.magnetic_field_strength.mean()
-#     expected_gyrofrequency = (s.q * mean_B / s.m).to(1 / u.s)
-#     expected_gyroradius = perp_speed / expected_gyrofrequency
-#     expected_gyroperiod = 2 * np.pi / expected_gyrofrequency
 
-#     dt = expected_gyroperiod / 100
+def test_particle_uniform_magnetic():
+    r"""
+        Tests the particle stepper for a uniform magnetic field motion.
+    """
+    # @profile
+    def magnetic_field(r):
+        return u.Quantity([[0, 0, 1]]*len(r), u.T)
 
-#     s = ParticleTracker(test_plasma, particle_type=particle_type, dt=dt, nt=int(1e4))
-#     s.v[:, 1] = perp_speed
+    # @profile
+    def electric_field(r):
+        return u.Quantity(np.zeros(r.shape), E_unit)
 
-#     s.v[:, 2] = parallel_speed
-#     s.run()
+    test_plasma = AnalyticalPlasma(magnetic_field, electric_field)
 
-#     x = s.position_history[:, 0, 0]
-#     z = s.position_history[:, 0, 2]
+    particle_type = 'N-14++'
+    s = ParticleTracker(test_plasma, particle_type=particle_type, dt=1e-2 * u.s,
+                        nt=int(1e2))
 
-#     try:
-#         params, stds = fit_sine_curve(x, s.t, expected_gyrofrequency)
-#     except RuntimeError as e:
-#         print(s)
-#         raise e
-#     estimated_gyroradius = np.abs(params[0]) * u.m
-#     estimated_gyroradius_std = np.abs(stds[0]) * u.m
-#     estimated_gyrofrequency = np.abs(params[1]) / u.s
-#     estimated_gyrofrequency_std = np.abs(stds[1]) / u.s
+    perp_speed = 0.01 * u.m / u.s
+    parallel_speed = 1e-5 * u.m / u.s
+    mean_B = 1 * u.T
+    expected_gyrofrequency = (s.q * mean_B / s.m).to(1 / u.s)
+    expected_gyroradius = perp_speed / expected_gyrofrequency
+    expected_gyroperiod = 2 * np.pi / expected_gyrofrequency
 
-#     assert np.isclose(expected_gyroradius, estimated_gyroradius,
-#                       atol=estimated_gyroradius_std), \
-#         "Gyroradii don't match!"
+    dt = expected_gyroperiod / 100
 
-#     assert np.isclose(expected_gyrofrequency, estimated_gyrofrequency,
-#                       atol=estimated_gyrofrequency_std), \
-#         "Gyrofrequencies don't match!"
+    s = ParticleTracker(test_plasma, particle_type=particle_type, dt=dt, nt=int(1e4))
+    s._v[:, 1] = perp_speed.si.value
```

Comment:
I had some issues with directly modifying `s.v[:, 1] = perp_speed`, still figuring that one out

Reply:
oh crud, https://stackoverflow.com/questions/3137685/using-property-decorator-on-dicts I think this is gonna be basically impossible